src/background/Wallet/Wallet.ts

```diff
@@ -1297,7 +1298,11 @@ export class Wallet {
         );
         return safeTxPlain;
       } catch (error) {
-        throw getEthersError(error);
+        const parsedError = getEthersError(error);
+        const parsed = parseError(parsedError);
```

Comment:
uhm :)

improve naming

Reply:
Ah, yes, good comment!
Will need to find a better name

Reply:
can simply do:

```
        const parsed = parseError(getEthersError(error));
```

Reply:
ah, i see you use it later, too